hpx/lcos/future.hpp

```diff
@@ -1117,17 +1117,25 @@ namespace hpx { namespace lcos
         typename traits::future_traits<shared_future>::result_type
         get(error_code& ec) const //-V659
         {
+            typedef
+                typename traits::future_traits<shared_future>::result_type
+                result_type;
             if (!this->shared_state_)
             {
                 HPX_THROWS_IF(ec, no_state,
                     "shared_future<R>::get",
                     "this future has no valid shared state");
-                return detail::future_value<R>::get_default();
+                static result_type tmp(detail::future_value<R>::get_default());
```

Comment:
Wouldn't it be cleaner to just make `get_default` return a const reference to static?